### ISSUE TITLE
the result of custom_RMSE is not same with default objective 'reg:linear'

### ISSUE THREAD

**Issue by 123infinity23** [dmlc/xgboost#3063]
Hi there,

I'm new to xgbost, and would like to adapt the objective function in R for a regression problem. I've read through everything that I could find on this (threads, presentations, docu, etc.), but still don't seem to get it right.

For testing purposes, I'v started with this simple case: 

1. Wright my on RMSE objective function
```
my_RMSEobj <- function(preds,dtrain){ 
   labels = getinfo(dtrain, "label")
 
  # test: RMSE
  grad <- (2./length(labels)) * (preds-labels)
  hess <- rep((2./length(labels)), length(grad))
  
  return(list(grad = grad, hess = hess))
} 
```

2. Run xgboost with custom objective function

```
param <- list("objective" = my_RMSEobj)
fit_xgb <- xgboost(param =param, data = train[, -c(1,6)], label = train[, c(6)], nrounds=nround, max_depth = 4, eta = 1, nthread = 2)
```

3. Re-run with reg:linear, and compare errors to the one received in step 2.

```
param <- list("objective" = 'reg:linear')
fit_xgb <- xgboost(param =param, data = train[, -c(1,6)], label = train[, c(6)], nrounds=nround, max_depth = 4, eta = 1, nthread = 2)
```

Output:
from Step 2: (only first 3 rounds)

```
[1]	train-rmse:39.564800 
[2]	train-rmse:38.696026 
[3]	train-rmse:38.598289 
```

from Step 3: (only first 3 rounds)

```
[1]	train-rmse:37.131550 
[2]	train-rmse:36.713436 
[3]	train-rmse:35.822300 
```

What did I do wrong when defining the objective function? Both should be the same functions, so the output shpuld be absolutely the same!

This really drives me nuts, so any help would be highly appreciated!!

Many thanks!!!

**Comment by khotilov** [comment]
Do not use the `(2./length(labels))` factor when computing gradients and hessians. You might be making them too small.
